sdk/core/core-http/src/serializer.ts

```diff
@@ -482,7 +516,8 @@ function serializeSequenceType(
   mapper: SequenceMapper,
   object: any,
   objectName: string,
-  isXml: boolean
+  isXml: boolean,
+  options?: { xmlCharKey?: string }
```

Comment:
feels like this private method shouldn't have `xmlCharKey` be optional, since the caller already knows the final value

Reply:
RIGHT! Making it required for private methods reveals some places where I missed passing the options

Reply:
oh, you meant requiring `xmlCharKey`.  I made `options` required.

After I replaced `{ xmlCharKey?: string }` with an interface, I would need some type mapping to make this property required, and assert it when calling I think.

Reply:
Made internal options `Required<>` as @xirzec suggested. I also rename the interface name to more generic `SerializerOptions`.